documentation/manual/src/main/asciidoc/modules/mongodb.asciidoc

```diff
@@ -1563,6 +1563,47 @@ Wife collection as JSON in MongoDB
 ----
 ====
 
+.Bidirectional one-to-one (Mapping to the same Entity)
+====
+[source, JAVA]
+----
+@Entity
+public class Person {
+
+    @Id
+    private String id;
+    private String name;
+
+    @OneToOne
+    private Person spouse;
+
+    // getters, setters ...
+}
+----
+
+Person (husband) collection as JSON in MongoDB
+
+[source, JSON]
+----
+{
+  "_id" : "alex",
+  "name" : "Alex",
+  "spouse" : "bea"
+}
+----
+
+Person (wife) collection as JSON in MongoDB
+
+[source, JSON]
+----
+{
+  "_id" : "bea",
+  "name" : "Bea",
+  "spouse" : "alex"
+}
+----
```

Comment:
Considering the previous comment I would rewrite this as:
```
Person  collection as JSON in MongoDB
[source, JSON]
----
{  "_id" : "Alex", role: "husband", "spouse" : "Bea" }
{  "_id" : "Bea",  role: "wife", "spouse" : "Alex" }
----
```

This make it clearer that they are part of the same collection.

Reply:
We need to also add a one-to-many association and show an example for that scenario as well.
Maybe, you can add an attribute children or similar to the same class and see if it doesn't get too confusing  to keep everything in one place. Alternatively, we can add a second example. Your choice

Reply:
Added children and made example simpler by removing the roles completely - spouse